### PR TITLE
Initial CP for unique identifier for association instance.

### LINKED ISSUE #99: Use OBR.3 and OBR.29 to identify device association for any action (association, disassociate, correction, wrong, update)  in DEV-52 messages
**Section Number** *Identify the most specific section number the issue occurs (e.g. 4.1.2)*

**Issue** *Describe your issue. Don't write a book, but do include enough to indicate what you see as a problem.*

**Proposed Change** *Propose a resolution to your issue (e.g., suggested new wording or description of a way to address the issue). The committee might simply accept your suggested text. Even if they don't, it gives a good sense of what you are looking for. Leaving this blank means you can't imagine how to resolve the issue, which makes it easier for the committee to admit they can't imagine how to resolve it either and leave it unresolved.*

**Priority**:
- High: Important issue where there is major issue to be resolved. Requires discussion and debate.
- Medium: Significant issue or clarification. Requires discussion, but should not lead to long debate.
- Low: Typo or other minor classification that an editor can manage. Requires no group discussion.

### PR DESCRIPTION
**Pull request by eldonmetz**
OBR-3 and OBR-29 usage similar to ACM PCD-[04,06 and 07].

Closes #99 

## 📑 Description
After PCIM connectathon testing in 2024, it was noted that it would simplify update logic for consumers if the Device-Patient Association Manager (DPAM) assigned a unique identifier at the start of an association instance and referenced it for updates. This would also be consistent with ACM profile PCD-04, PCD-06 and PCD-07 transactions.

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed
- [ ] I have selected a committee co-chair to review the PR

### CODE REVIEW

File: asciidoc/cp_nn6.adoc [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+|IHE Technical Framework or Supplement modified:
+|PCIM Profile TI revision 2.0, dated 2024-04-11
+
+|Volume(s) and Section(s) affected:
+|Trial Implementation, Multiple Sections
+
+2+|Rationale for Change:
+
+After PCIM connectathon testing in 2024, it was noted that it would simplify update logic for consumers if to the Device-Patient Association Manager (DPAM) assigned a unique identifier at the start of an association instance and referenced it for updates. This would also be consistent with ACM profile PCD-04, PCD-06 and PCD-07 transactions.
```
Comment (engelbert65) [inline asciidoc/cp_nn6.adoc:57]: Typo:  "...for consumers **if to** the Device-Patient..." should be "...for consumers **if** the Device-Patient..."

File: asciidoc/cp_nn6.adoc [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+|===
+
+| _Section B.7 OBR - Observation Reqeust Segment_  *OBR-3 Filler Order Number* , add the following definition after the last paragraph ending on line 2482:
+
+|===
+[.text-left]
+[underline]#Proposed Text:#
+
+[.text-left]
+In the DEV-51 and DEV-52 transactions of the PCIM Profile, this field serves as the unique identifier for the initial association instance and each subsequent update to that association. This value is assigned by the Device-Patient Association Reporter and Device-Patient Association Manager. For the initial association message, OBR-29 Parent is required to be empty.  For all subsequent updates to the same association, OBR-3 Filler Order Number identifies the unique update and OBR-29 Parent contains the value from OBR-3 Filler Order Number of the initial association instance. Device-Patient Association Reporters that are sending an update for an association instance they did not create are not required to fill in OBR-29, but Device-Patient Association Managers must always fill in OBR-29 for all updates. This allows the Device-Patient Association Consumer to correlate all subsequent updates with the original association.
```
Comment (engelbert65) [inline asciidoc/cp_nn6.adoc:72]: Need to make it clear that OBR-3 from the initial association message goes in OBR-29.2 (Parent.Filler Assigned Identifier), and that the delimiter within OBR-3 changes from '^' to '&' when putting it in OBR-29.2.

Here's text from an ACM CP written 2 years ago (https://docs.google.com/document/d/185t8bjqTm9ZF4sSxp9aGYFFozc6orSLu/edit?pli=1):

"...a transcription of the OBR-3 component values to subcomponents while replacing the component delimiters (^) with subcomponent delimiters (&)"

File: asciidoc/cp_nn6.adoc [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+| _Section B.7 OBR - Observation Request Segment_  *OBR-29 Parent (EIP) 00261* , add the following definition after the last paragraph ending on line 2590:
+
+|===
+[.text-left]
+[underline]#Proposed Text:#
+
+[.text-left]
+This field serves as the unique association instance identifier in DEV-51 and DEV-52 transactions of the PCIM profile. It is assigned by a Device-Patient Association Reporter and Device-Patient Association Manager, and is used by Device-Patient Association Manager and Consumers to associate all updates to a particular association instance throughout the history of the association. The Device-Patient Association Reporter may fill in OBR-29 if it also reported the initial association instance. This is optional, as there may be multiple reporters associating and disassociating devices. The Device-Patient Association Manager shall fill in OBR-29 and the same value will be sent to Device-Patient Association Consumers in all updates to an association instance. See OBR-3 Filler order Number for additional information.
+
```
Comment (engelbert65) [inline asciidoc/cp_nn6.adoc:84]: Same as above, need to be clear that OBR-29.2 is the value of interest.

File: asciidoc/cp_nn6.adoc [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+| 8
+| TS
+| C
+| Latest participant involvement
+
+| 29
+| EIP
+| C
+| Unique instance identifier for the originating association event that all updates apply to. See Appendix B.7 OBR - Observation Request Segment OBR-29 Parent (EIP) 00261for additional details.
```
Comment (engelbert65) [inline asciidoc/cp_nn6.adoc:163]: Typo:  add space after "00261".

**Review by engelbert65 — CHANGES_REQUESTED** [review]

**Comment by eldonmetz** [comment]
@engelbert65 Changes made based on feedback. [HTML output here.](https://htmlpreview.github.io/?https://github.com/IHE/DEV.PCIM/blob/dd79aa9684578fedc2ebd16aa7c3e4de93d5aef7/CPs/in_progress/cp_nn6.html)

**Comment by eldonmetz** [comment]
@engelbert65  Additional formatting changes. [HTML here.](https://htmlpreview.github.io/?https://github.com/IHE/DEV.PCIM/blob/f6999376051775afc6769dfa2a59a0d307c633d4/CPs/in_progress/cp_nn6.html)

File: asciidoc/cp_nn6.adoc [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+In the DEV-51 and DEV-52 transactions of the PCIM Profile, this field serves as the unique identifier for the initial association instance and each subsequent update to that association. This value is assigned by the Device-Patient Association Reporter (DEV-51) and Device-Patient Association Manager (DEV-52). For the initial association message, OBR-29 Parent is required to be empty.  For all subsequent updates to the same association, OBR-3 Filler Order Number identifies the unique update and OBR-29.2 Parent Filler-Assigned Identifier contains the value from OBR-3 Filler Order Number of the initial association instance. Device-Patient Association Reporters that are sending an update for an association instance they did not create are not required to fill in OBR-29.2, but Device-Patient Association Managers shall fill in OBR-29.2 for updates. This allows the Device-Patient Association Consumer to correlate all subsequent updates with the original association. 
+
+[.text-left]
+When transcribing the the components of the OBR-3 field value to the OBR-29.2, all subscomponents shall be transcribed and the component delimiters (^) replaced with subcomponent delimiters (&).
```
Comment (engelbert65) [inline asciidoc/cp_nn6.adoc:75]:
```suggestion
When transcribing the the components of the OBR-3 field value to OBR-29.2, all subcomponents shall be transcribed and the component delimiters (^) replaced with subcomponent delimiters (&).
```

File: asciidoc/cp_nn6.adoc [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+|===
+[.text-left]
+[underline]#Proposed Text:#
+
+[.text-left]
+The EI data type value in the OBR-29.2 Parent Filler-Assigned Identifier field serves as the unique association instance identifier in DEV-51 and DEV-52 transactions of the PCIM profile. It is assigned by a Device-Patient Association Reporter (DEV-51) and Device-Patient Association Manager (DEV-52), and is used by Device-Patient Association Manager and Device-Patient Consumer to associate all updates to a particular association instance throughout the history of the association. The Device-Patient Association Reporter shall fill in OBR-29.2 with the originating OBR.3 Filler Order Number for all updates to association instances that it initially originated. The Device-Patient Association Manager shall fill in OBR-29.2 with the originating OBR.3 Filler Order Number in all DEV-52 messages conveying updates to an association instance. See OBR-3 Filler order Number for additional information.
+
+[.text-left]
+When transcribing the components of the OBR-3 field value to the OBR-29.2 all subcomponents shall be transcribed and the component delimiters (^) replaced with subcomponent delimiters (&).
```
Comment (engelbert65) [inline asciidoc/cp_nn6.adoc:88]:
```suggestion
When transcribing the components of the OBR-3 field value to OBR-29.2, all subcomponents shall be transcribed and the component delimiters (^) replaced with subcomponent delimiters (&).
```

File: asciidoc/cp_nn6.adoc [hunk 7]
```diff
@@ -170,17 +170,51 @@ When transcribing the components of the OBR-3 field value to OBR-29.2, all subco
...
+PRT|1|UC||EQUIP|||||3 WEST ICU^3001^1|MON5596^^231A8456B1CB2366^EUI-64|20160726160000
+PRT|2|UC||AUT|58796^Ratched^N||||3 WEST ICU^3001^1||20160726160000
+....
+
+
+[.text-left]
+[underline]#Proposed Text:#
+[.text-left]
+....
+MSH|^~\&|MonitorGateway||AssocMgr||20160726160000||ORU^R01^ORU_R01|12d1574|P|2.6|||AL|AL|USA||||IHE_DEV_051^IHE PCD\^1.3.6.1.4.1.19376.1.6.1.51.1^ISO
```
Comment (engelbert65) [inline asciidoc/cp_nn6.adoc:194]: Remove the backslash after "IHE PCD"

File: asciidoc/cp_nn6.adoc [hunk 8]
```diff
@@ -170,17 +170,51 @@ When transcribing the components of the OBR-3 field value to OBR-29.2, all subco
...
+
+|===
+[.text-left]
+[underline]#Existing Text:#
+[.text-left]
+....
+MSH|^~\&|AssocMgr||AssocConsumer||20160726160000||ORU^R01^ORU_R01|12d1599|P|2.6|||AL|AL|USA||||IHE_DEV_052^IHE PCD^1.3.6.1.4.1.19376.1.6.1.52.1^ISO
+PID|||AB60001^^^A^PI||McMurphy^R^P^^^^L
+PV1||E|3 WEST ICU^3001^1
+OBR|||15404697^^|69136^MDC_OBS_ASSOCIATION_PATIENT_DEVICE^MDC|||20160726160000|20160726160000
```
Comment (engelbert65) [inline asciidoc/cp_nn6.adoc:215]: The "^^" in OBR-3 are not in the existing text.

**Review by engelbert65 — APPROVED** [review]

**Pull request merged**